trinity/utils/filesystem.py

```diff
@@ -0,0 +1,22 @@
+import os
+
+
+def is_same_path(p1, p2):
+    n_p1 = os.path.abspath(os.path.expanduser(p1))
+    n_p2 = os.path.abspath(os.path.expanduser(p2))
+
+    try:
+        return os.path.samefile(n_p1, n_p2)
+    except FileNotFoundError:
+        return n_p1 == n_p2
+
+
+def is_under_path(base_path, path, strict=True):
+    if is_same_path(base_path, path):
+        if strict:
+            return False
+        else:
+            return True
+    absolute_base_path = os.path.abspath(base_path)
+    absolute_path = os.path.abspath(path)
```

Comment:
Probably want to `expanduser` here, too?

Looks like a `_normalize_path = compose(os.path.abspath, os.path.expanduser)` might be useful in this module.